### PR TITLE
feat: support preact

### PR DESCRIPTION
**Pull request by OSpoon**
Hi~ I have added `preact` and `preact-typescript` template folders to `programs/create/templates` and made some necessary configuration changes. The only downside is that it might not be possible to use [Functional Components](https://preactjs.com/guide/v10/components#functional-components). Other than that, the experience should be consistent with the `react` and `react-typescript` templates.

I will also continue to investigate the real reason why functional components cannot be used.

### CODE REVIEW

File: programs/create/templates/preact/template/newtab/base.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+h1 {
+  font-size: 4.7em;
+}
+
+.preact {
+  animation: spin 5s linear infinite;
+}
+
+@keyframes spin {
```
Comment (cezaraugusto) [inline programs/create/templates/preact/template/newtab/base.css:16]: do you mind changing this animation to something else? otherwise just make it static

**Review by cezaraugusto — COMMENTED** [review]
Hi @OSpoon! This is another great one! Comments left

Regarding the functional components: I managed to get the new tab page example running by enabling `ReactRefreshPlugin`. Maybe adding a Preact conditional in there might do the trick? Relevant file: https://github.com/cezaraugusto/extension.js/blob/main/programs/develop/webpack/plugins/reloadPlugins.ts#L19

File: programs/create/templates/preact/template/newtab/base.css (hunk 1 above)
Comment (OSpoon) [reply programs/create/templates/preact/template/newtab/base.css]: I am now using the same animations as [Preact](https://preactjs.com/).

File: programs/create/templates/preact-typescript/template/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# [projectName]
+
+> This project was bootstrapped using the Extension.js React-TypeScript template.
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/create/templates/preact-typescript/template/README.md]: Thank you for reminding me.

File: programs/create/templates/preact/template/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# [projectName]
+
+> This project was bootstrapped using the Extension.js TypeScript template.
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/create/templates/preact/template/README.md]: I agree with your review.

**Review by cezaraugusto — APPROVED** [review]
Looking 💯 great work!! ++

**Pull request merged**